checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleSeverityUtils.java

```diff
@@ -34,18 +34,22 @@ private CheckstyleSeverityUtils() {
     }
 
     public static String toSeverity(RulePriority priority) {
+        return toSeverity(priority.name());
+    }
+
+    public static String toSeverity(String priority) {
         final String result;
 
         switch (priority) {
-            case BLOCKER:
-            case CRITICAL:
+            case "BLOCKER":
```

Comment:
There is no way to convert String priority into `RulePriority` to leave the enumeration switch?

Reply:
Well, there's another issue with `RulePriority` -> it's deprecated since 4.2. The new `ActiveRule` implementation only returns the String value. I actually don't know if the annotations of SQ make sense because they deprecate `RulePriority` but don't deprecate `ActiveRule::getSeverity` (server side) which in turn returns `RulePriority` but they deprecated the method `getPriority`.
Hence, I see two options: we convert the String from the new `ActiveRules` to the deprecated `RulePriority` and leave the enumeration switch or we use a String switch but, personally, I'd prefer not using deprecated classes?

Reply:
Ok, ignore this for now.
Is there any chance to remove this deprecated usage then?